### ISSUE TITLE
Misplaced Spawnpoint

### ISSUE THREAD

**Issue by Infernus101** [pmmp/PocketMine-MP#3639]
### Issue description

- Expected result: Teleport to the actual spawnpoint
- Actual result: What actually happened? Teleports to 0, 255, 0 or below the spawnpoint (in void)

### Steps to reproduce the issue
```
$warp = $this->getServer()->getDefaultLevel()->getSafeSpawn();
$warp->getLevel()->loadChunk($warp->x >> 4, $warp->z >> 4);
$player->teleport($warp);
```
using the above code in PlayerLoginEvent. Resetting the spawnpoint didn't work.

### OS and versions

* PocketMine-MP: 3.14.0
* PHP: 7.3
* Server OS: Linux
* Game version: Win10

### Plugins

- If you remove all plugins, does the issue still occur?
- If the issue is **not** reproducible without plugins:
  - Have you asked for help on our forums before creating an issue?
  - Can you provide sample, *minimal* reproducing code for the issue? If so, paste it in the bottom section

**Comment by Infernus101** [comment]
Fixed in 4ae3fd7

**Comment by dktapps** [comment]
That's impossible. That commit affects behaviour **after** PlayerLoginEvent. It has no effect on PlayerLoginEvent whatsoever.
